File: CS50/C/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int hash(int n);
int space(int n);

int main(void)
{
    //Promting the user for the height but allowing them only using numbers from 1 to 8
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while((height < 1) || (height > 8));

    int spaces = height;

    //Printing the pyramid as the height the user wants
    for(int i = 1; i <= height; i++)
    {
        spaces--;
        space(spaces);
        hash(i);
        printf("  ");
        hash(i);
        printf("\n");
    }
}



int hash(int n)
{
    for(int i = 0; i < n; i++)
    printf("#");

    return 0;
}

int space(int n)
{
    for(int j = 0; j < n; j++)
    printf(" ");

    return 0;
}

File: CS50/Python/mario/mario.py
"""
 mario.py
 Program that prints a mario pyramid
 By Pol
 Created 22/04/2022
"""


def main():
    # Promting the user for the pyramids' size
    while True:
        try:
            height = int(input("Height: "))
        except ValueError:
            continue
        if (height > 0) and (height < 9):
            break

    # Printing the pyramid
    tmp = height
    for i in range(height):
        space(tmp)
        block(i)
        print("  ", end="")
        block(i)
        print()
        tmp -= 1


# Creating a function that prints block/s
def block(size):
    print("#" * (size + 1), end="")


# Creating a function that prints spaces
def space(size):
    print(" " * (size - 1), end="")


# Running main
main()
